app/controllers/assignments_controller.rb

```diff
@@ -269,6 +269,20 @@ def summary
     end
   end
 
+  def download_test_results
+    @assignment = Assignment.find(params[:id])
+    respond_to do |format|
+      format.json do
+        data = @assignment.summary_test_result_json(@current_user).to_json
```

Comment:
Move the `.to_json` call into the assignment method, since the method name suggests it returns JSON.